apps/src/javalab/ControlButtons.jsx

```diff
@@ -47,11 +66,11 @@ export default function ControlButtons({
         <JavalabSettings>{renderSettings()}</JavalabSettings>
         {!isEditingStartSources && (
           <JavalabButton
-            text={i18n.finish()}
-            onClick={onContinue}
+            text={finishButtonText}
+            onClick={isSubmittable ? null : onContinue}
```

Comment:
In https://github.com/code-dot-org/code-dot-org/blob/47be99d6cf7df2be746b592906f50c0f3860b80a/apps/src/submitHelper.js#L26, an onclick handler is added to this button. We don't want to add a second here or it will try to reload the page and submit the level at the same time.